### PR TITLE
New-style Alerts

### PR DESCRIPTION
**Pull request by zdeyn**
I've provided three things here:

- a set of styles, with associated graphics, frame objects and delay
- the function `Hardcore:ShowAlertFrame(style, message)`, which accepts:
  - a string containing the name of a style, and
  - a string containing the message to be displayed as per the style
- a slashcommand, `/hc alert style message in multiple words goes here` which pops up an alert of the appropriate style and message

There is a known issue multiple alerts being called in quick successful - the timer closes the last one, instead of fading between them

The code isn't very 'lua-esque', I'm still learning the local idioms, and would appreciate any code revisions or suggestions

### CODE REVIEW

**Comment by zdeyn** [comment]
This now includes Brandung's tweaks, although it could use another eye to ensure that changes made post-justification are still on-point

File: Hardcore.lua [hunk 1]
```diff
@@ -471,6 +537,37 @@ function Hardcore:Debug(msg)
...
+	icon:SetTexture(filename)
+	text:SetText(message)
+
+	frame:Show()
+
+	-- TODO: Allow custom sounds per-frame, or allow passing a sound to the function
+	PlaySound(8959)
+
+	-- HACK:
+	-- There's a bug here where a sequence of overlapping notifications share one 'hide' timer
```
Comment (blenderer) [inline Hardcore.lua:564]: Possibly assign the ticker to a global and cancel it when a new one opens. Not super important now though

File: Hardcore.xml [hunk 2]
```diff
@@ -62,6 +84,32 @@
 		</Layers>
   </Frame>
 
+  <?comment msg="This provides dynamic style templating, via ALERT_STYLES" ?>
```
Comment (blenderer) [inline Hardcore.xml:87]: 👍

**Review by blenderer — APPROVED** [review]
Nice stuff!

**Comment by RobertB4** [comment]
Sorry I just merged your other PR, didn't think it would cause a conflict 😕 
This PR should be good to merge once the conflict is fixed.

**Review by blenderer — COMMENTED** [review]
Can we get some screenshots in the PR description too?

**Comment by zdeyn** [comment]
Some rough pics to capture current status:

image

image

image


We should be able to add new icons, frame backgrounds, custom sounds, fonts and strings any time we like from here

**Review by RobertB4 — APPROVED** [review]

**Comment by zdeyn** [comment]
Two approvals (blenderer and RobertB4), so merging in.. Tested on Era

**Pull request merged**